### DISCUSSION TITLE
Auto Restart Android App

### DISCUSSION THREAD

**Discussion by zburlott4** [airsdk/Adobe-Runtime-Support#3093] [General]
Is there anyway to auto restart an app that crashes? One of our apps periodically crashes and we want to automatically restart the app until we can determine to root cause. We have found the below link, is there anything within flash that would allow us to do this?

https://stackoverflow.com/questions/2681499/android-how-to-auto-restart-an-application-after-it-has-been-force-closed

**Comment by marchbold** [comment]
I've never tried this but that `setDefaultUncaughtExceptionHandler` method is what the Exceptions ANE implements. We could probably slightly modify it to start the app again. Not sure this is considered best behaviour though and may see your app review get rejected?

**Reply by raresn** [reply]
@marchbold we had the need in the past to restart apps on android on some internal projects (not launch on playstore). We worked around it, but i think it's nice to have if it's easy to implement.

**Reply by pol2095** [reply]
You can try this [ANE](http://pol2095.free.fr/NativeApp/), if you manage to catch the error but it isn't a good behavior as @Marchbold mentioned.

**Comment by zburlott4** [comment]
@raresn how did you do the app restart in the past. Did you require a native extension to do this?
